File: qc-core/src/main/java/com/eager/questioncloud/core/domain/post/implement/PostCommentReader.java
package com.eager.questioncloud.core.domain.post.implement;

import com.eager.questioncloud.core.common.PagingInformation;
import com.eager.questioncloud.core.domain.post.dto.PostCommentDetail;
import com.eager.questioncloud.core.domain.post.repository.PostCommentRepository;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class PostCommentReader {
    private final PostCommentRepository postCommentRepository;
    private final PostValidator postValidator;

    public List<PostCommentDetail> getPostComments(Long postId, Long userId, PagingInformation pagingInformation) {
        postValidator.permissionValidator(postId, userId);
        return postCommentRepository.getPostCommentDetails(postId, userId, pagingInformation);
    }

    public int count(Long postId) {
        return postCommentRepository.count(postId);
    }
}

File: qc-core/qc-post/qc-post-api/src/main/kotlin/com/eager/questioncloud/post/implement/PostCommentReader.kt
package com.eager.questioncloud.post.implement

import com.eager.questioncloud.common.pagination.PagingInformation
import com.eager.questioncloud.post.dto.PostCommentDetail
import com.eager.questioncloud.post.repository.PostCommentRepository
import com.eager.questioncloud.user.api.internal.UserQueryAPI
import org.springframework.stereotype.Component

@Component
class PostCommentReader(
    private val postCommentRepository: PostCommentRepository,
    private val userQueryAPI: UserQueryAPI,
) {
    fun getPostCommentDetails(
        userId: Long,
        postId: Long,
        pagingInformation: PagingInformation
    ): List<PostCommentDetail> {
        val postComments = postCommentRepository.findByPostIdWithPagination(postId, pagingInformation)
        val writerIds = postComments.map { it.writerId }
        val writers = userQueryAPI.getUsers(writerIds).associateBy { it.userId }
        
        val postCommentDetails = mutableListOf<PostCommentDetail>()
        
        for (postComment in postComments) {
            val writer = writers.getValue(postComment.writerId)
            postCommentDetails.add(
                PostCommentDetail(
                    postComment.id,
                    writer.name,
                    writer.profileImage,
                    postComment.comment,
                    postComment.isCreator,
                    writer.userId == userId,
                    postComment.createdAt
                )
            )
        }
        
        return postCommentDetails
    }
    
    fun count(postId: Long): Int {
        return postCommentRepository.count(postId)
    }
}